cmd/boulder-wfe2/main.go

```diff
@@ -62,6 +69,75 @@ type config struct {
 	}
 }
 
+// loadCertificateChains processes the provided chainConfig of AIA Issuer URLs
+// and cert filenames. For each AIA issuer URL all of its cert filenames are
+// read, validated as PEM certificates, and concatinated together separated by
+// newlines. The combined PEM certificate chain contents for each are returned
+// in the results map, keyed by the AIA Issuer URL.
+func loadCertificateChains(chainConfig map[string][]string) (map[string]string, error) {
+	results := make(map[string]string, len(chainConfig))
+
+	// For each AIA Issuer URL we need to read the chain files
+	for aiaIssuerURL, chainfiles := range chainConfig {
+		var buffer bytes.Buffer
+
+		// There must be at least one chainfile specified
+		if len(chainfiles) == 0 {
+			return nil, berrors.MalformedError(
+				"CertificateChain entry for AIA issuer url %q has no chain "+
+					"file names configured",
+				aiaIssuerURL)
+		}
+
+		// Read each chain file - it is expected to be a PEM certificate
+		for _, c := range chainfiles {
+			// Read the raw contents
```

Comment:
This `for` loop is a little long. Could you factor it out into its own function? That function could return `[]byte, err`, and the caller could use `bytes.Join` to join the result.

Reply:
> This for loop is a little long. Could you factor it out into its own function? That function could return []byte, err

Sure :+1: 

>  the caller could use bytes.Join to join the result.

We also want a trailing delimiter so I don't think `bytes.Join` works. I'd still have to tack on a `\n` at the end. I'd rather leave it the way it is I think.

Reply:
Implemented in a394b4d